authenticator/server/credmgrs/hc_vault.go

```diff
@@ -72,20 +75,23 @@ func (h *hcVaultCredMgr) Password(_ *log.Entry, identity DBKey) (string, error)
 // vaultClient retrieves a client using either the environmental VAULT_TOKEN,
 // or reading the latest token from the token file sink.
 func (h *hcVaultCredMgr) vaultClient() (*vault.Client, error) {
-	// Only use the token sink if there's not already an environmental
-	// VAULT_TOKEN.
-	if h.tokenPath != "" && os.Getenv(vault.EnvVaultToken) == "" {
+	// Only use the token sink if a vault token is not provided
+	if h.tokenPath != "" && h.token == "" {
 		tokenBytes, err := ioutil.ReadFile(h.tokenPath)
 		if err != nil {
 			return nil, err
 		}
-		// There is no way to directly pass in the token, so we
-		// must set it in the environment.
-		os.Setenv(vault.EnvVaultToken, string(tokenBytes))
-		defer os.Unsetenv(vault.EnvVaultToken)
+		h.token = string(tokenBytes)
 	}
 
 	// This uses a default configuration for Vault. This includes reading
 	// Vault's environment variables and setting them as a configuration.
-	return vault.NewClient(nil)
+	config := vault.DefaultConfig()
```

Comment:
I love how you handled letting them pass in the address and token here! Very nice how you parse the env vars first, and then add those on.